internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go

```diff
@@ -207,6 +207,11 @@ func (k *KafkaManager) calculateAvailableAndMaxCapacityForDynamicScaling(streami
 	// of Kafkas have already been created. In this case we limit the metric to 0 anyways.
 	available := math.Max(0, maxAvailable-float64(streamingUnitsByRegion.Count))
 
+	if streamingUnitsByRegion.ClusterType == api.EnterpriseDataPlaneClusterType.String() {
+		maxAvailable = float64(streamingUnitsByRegion.MaxUnits)
+		available = maxAvailable - float64(streamingUnitsByRegion.MaxUnits)
+	}
```

Comment:
As discussed, when we encounter an enterprise cluster, we shouldn't compute the region limits at all: this is current happening in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/11d2609bfca4435ded886e1899c68e438a96bae6/internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go#L210